File: memory-api/src/main/java/com/memory/dto/hashtag/response/HashTagResponse.java
package com.memory.dto.hashtag.response;

import com.memory.domain.hashtag.HashTag;

public record HashTagResponse(
        Long id,
        String name,
        Long useCount
) {

    public static HashTagResponse from(HashTag hashTag) {
        return new HashTagResponse(
                hashTag.getId(),
                hashTag.getName(),
                hashTag.getUseCount()
        );
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/hashtag/response/HashTagResponse.kt
package com.memory.dto.hashtag.response

import com.memory.domain.hashtag.HashTag

data class HashTagResponse(
    val id: Long?,
    val name: String?,
    val useCount: Long?
) {
    companion object {
        @JvmStatic
        fun from(hashTag: HashTag): HashTagResponse {
            return HashTagResponse(
                hashTag.id,
                hashTag.name,
                hashTag.useCount
            )
        }
    }
}